packages/ui/src/SliderFormik.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ReactElement, useCallback } from 'react'
+import { FieldValidator, useField } from 'formik'
+import { Slider, SliderExtraProps } from './Slider'
+
+export type SliderFormikProps<V extends object> = SliderExtraProps & {
+  name: keyof V
+  validate?: FieldValidator
+}
+
+export const SliderFormik = <V extends object>({ name, validate, ...props }: SliderFormikProps<V>): ReactElement => {
+  const [field, meta, { setValue, setTouched }] = useField<number | [number, number]>({
```

Comment:
What if we add an explicit `range?: boolean` prop to the `Slider`? Then we would be able to strictly type the value that we get here or the one that we get in the `onChange` callback.

```tsx
type Value = typeof range extends true ? [number,number] : number

useField<Value>(..)

onChange = useCallback((newValue: Value) => ...)
```

Reply:
@aigoncharov as we discussed on Slack last time, we will keep the current implementation with typeguards.